src/app/features/wallets/transfer/transfer.page.ts

```diff
@@ -0,0 +1,194 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Component } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { ActivatedRoute } from '@angular/router';
+import { NavController, Platform } from '@ionic/angular';
+import { TranslocoService } from '@ngneat/transloco';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { BehaviorSubject, combineLatest, of } from 'rxjs';
+import { catchError, concatMap, first, tap } from 'rxjs/operators';
+import { BlockingActionService } from '../../../shared/blocking-action/blocking-action.service';
+import { DiaBackendStoreService } from '../../../shared/dia-backend/store/dia-backend-store.service';
+import { DiaBackendWalletService } from '../../../shared/dia-backend/wallet/dia-backend-wallet.service';
+import { ErrorService } from '../../../shared/error/error.service';
+import { TransferLoadingComponent } from './transfer-loading/transfer-loading.component';
+import { TransferRequestSentComponent } from './transfer-request-sent/transfer-request-sent.component';
+
+@UntilDestroy({ checkProperties: true })
+@Component({
+  selector: 'app-transfer',
+  templateUrl: './transfer.page.html',
+  styleUrls: ['./transfer.page.scss'],
+})
+export class TransferPage {
+  mode = '';
+
+  readonly assetWalletBscNumBalance$ =
+    this.diaBackendWalletService.assetWalletBscNumBalance$;
+  readonly integrityWalletBscNumBalance$ =
+    this.diaBackendWalletService.integrityWalletBscNumBalance$;
+
+  transferAmount: number | null = null;
+
+  readonly gasFee$ = new BehaviorSubject<number>(0);
+  readonly totalCost$ = new BehaviorSubject<number>(0);
+
+  readyToSend = false;
+
+  keyboardIsShown = false;
+
+  private orderId = '';
+
+  constructor(
+    private readonly activeRoute: ActivatedRoute,
+    private readonly diaBackendWalletService: DiaBackendWalletService,
+    private readonly blockingActionService: BlockingActionService,
+    private readonly storeService: DiaBackendStoreService,
+    private readonly dialog: MatDialog,
+    private readonly navCtrl: NavController,
+    private readonly errorService: ErrorService,
+    private readonly translocoService: TranslocoService,
+    private readonly platform: Platform
+  ) {
+    this.activeRoute.paramMap.subscribe(paramMap => {
+      if (!paramMap.has('mode')) {
+        this.navCtrl.navigateBack('/wallets');
+        return;
+      }
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      this.mode = paramMap.get('mode')!;
```

Comment:
Yes the implementation you mentioned will work just fine here. I just thought that making `mode` an Observable is overkilled since 
1. Once users visit transfer page, `mode` will never change. 
2. This page will 100% needs to use this variable. It's not like there's some chance that we might not need it so let's load it lazily.

And so I just make it a regular class variable so that it's easier to work with.

Reply:
There is actually a possibility I thought of when I'm considering this. Many webpages or app with similar UI interfaces has a functionality: when user click the arrow button, they can revert the direction of transfer without refreshing or leaving the page. We don't have that kind of design so it's overkill for now, but that's also flexibility.

Anyway considering YAGNI, I think we can keep it like this and just change it when it really needs to be changed